core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -473,10 +472,12 @@ class WorkspaceService(protected val userInfo: UserInfo,
       // Delete workflowCollection resource in sam outside of DB transaction
       _ <- traceWithParent("deleteWorkflowCollectionSamResource", parentSpan)(_ =>
         workspaceContext.workflowCollectionName.map( cn => samDAO.deleteResource(SamResourceTypeNames.workflowCollection, cn, userInfo) ).getOrElse(Future.successful(())) recoverWith {
-          case t:Throwable => {
-            logger.error(s"Unexpected failure deleting workspace (while deleting workflowCollection in Sam) for workspace `${workspaceName}`", t)
+          case t: RawlsExceptionWithErrorReport if t.errorReport.statusCode.contains(StatusCodes.NotFound) =>
```

Comment:
As far as I can tell by looking at the [method called by `samDAO.deleteResources`](https://github.com/broadinstitute/rawls/blob/0e0b01b4b0d5bfaba79cd87b8e82ec36fa90e2e6/core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpSamDAO.scala#L48), we can assume that "normal" exceptions will be of type `RawlsExceptionWithErrorReport`.

Reply:
👍 makes sense, as mentioned, catching `Throwable` is rarely a good idea.